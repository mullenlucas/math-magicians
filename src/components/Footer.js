import React from 'react';

function Footer() {
  return (
    <footer>
      <p>Lucas Mullen - Â© 2022 - All rights reserved.</p>
    </footer>
  );
}

export default Footer;
