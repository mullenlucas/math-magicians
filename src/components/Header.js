import React from 'react';
import { NavLink } from 'react-router-dom';

function Header() {
  return (
    <header>
      <h1>Math Magicians</h1>
      <div id="h-links">
        <NavLink to="/" className="links">Home</NavLink>
        |
        <NavLink to="/calculator" className="links">Calculator</NavLink>
        |
        <NavLink to="/quote" className="links">Quote</NavLink>
      </div>
    </header>

  );
}

export default Header;
