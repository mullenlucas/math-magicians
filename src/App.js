import React from 'react';
import { Outlet } from 'react-router-dom';
import Header from './components/Header';
import Footer from './components/Footer';

function App() {
  return (
    <div className="App">
      <Header />
      <main id="main-section">
        <Outlet />
      </main>
      <Footer />
    </div>
  );
}

export default App;
