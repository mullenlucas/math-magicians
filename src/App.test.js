import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import App from './App';
import Header from './components/Header';
import Footer from './components/Footer';

describe('Testing App components', () => {
  test('Rendering the Header component', () => {
    const component = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });
  test('Rendering the App component', () => {
    const component = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });
  test('Rendering the Footer component', () => {
    const component = render(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });
});
