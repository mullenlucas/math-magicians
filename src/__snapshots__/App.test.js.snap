// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App components Rendering the App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header>
          <h1>
            Math Magicians
          </h1>
          <div
            id="h-links"
          >
            <a
              aria-current="page"
              class="links active"
              href="/"
            >
              Home
            </a>
            |
            <a
              class="links"
              href="/calculator"
            >
              Calculator
            </a>
            |
            <a
              class="links"
              href="/quote"
            >
              Quote
            </a>
          </div>
        </header>
        <main
          id="main-section"
        />
        <footer>
          <p>
            Lucas Mullen - © 2022 - All rights reserved.
          </p>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header>
        <h1>
          Math Magicians
        </h1>
        <div
          id="h-links"
        >
          <a
            aria-current="page"
            class="links active"
            href="/"
          >
            Home
          </a>
          |
          <a
            class="links"
            href="/calculator"
          >
            Calculator
          </a>
          |
          <a
            class="links"
            href="/quote"
          >
            Quote
          </a>
        </div>
      </header>
      <main
        id="main-section"
      />
      <footer>
        <p>
          Lucas Mullen - © 2022 - All rights reserved.
        </p>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing App components Rendering the Footer component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer>
        <p>
          Lucas Mullen - © 2022 - All rights reserved.
        </p>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer>
      <p>
        Lucas Mullen - © 2022 - All rights reserved.
      </p>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing App components Rendering the Header component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <h1>
          Math Magicians
        </h1>
        <div
          id="h-links"
        >
          <a
            aria-current="page"
            class="links active"
            href="/"
          >
            Home
          </a>
          |
          <a
            class="links"
            href="/calculator"
          >
            Calculator
          </a>
          |
          <a
            class="links"
            href="/quote"
          >
            Quote
          </a>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <h1>
        Math Magicians
      </h1>
      <div
        id="h-links"
      >
        <a
          aria-current="page"
          class="links active"
          href="/"
        >
          Home
        </a>
        |
        <a
          class="links"
          href="/calculator"
        >
          Calculator
        </a>
        |
        <a
          class="links"
          href="/quote"
        >
          Quote
        </a>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
